src/app_service/service/chat/service.nim

```diff
@@ -400,22 +400,41 @@ QtObject:
         chats.add(chat)
     result = (chats, messages)
 
-  proc processMessageUpdateAfterSend*(self: Service, response: RpcResponse[JsonNode]): (seq[ChatDto], seq[MessageDto]) =
-    result = self.parseChatResponse(response)
-    var (chats, messages) = result
+  proc parseGroupChatResponse*(self: Service, response: RpcResponse[JsonNode]): (ChatDto, seq[MessageDto]) =
+    var chat: ChatDto
+    var messages: seq[MessageDto] = @[]
+    if response.result{"chat"} != nil:
+      chat = chat_dto.toChatDto(response.result["chat"])
+      self.updateOrAddChat(chat)
+    if response.result{"messages"} != nil:
+      for jsonMsg in response.result["messages"]:
+        messages.add(jsonMsg.toMessageDto)
+    return (chat, messages)
+
+  proc signalChatsAndMessagesUpdates*(self: Service, chats: seq[ChatDto], messages: seq[MessageDto]) =
     if chats.len == 0 or messages.len == 0:
       error "no chats or messages in the parsed response"
       return
-
     for chat in chats:
       if (chat.active):
         self.events.emit(SIGNAL_CHAT_CREATED, CreatedChatArgs(chat: chat))
-
+    var chatMap: Table[string, ChatDto] = initTable[string, ChatDto]()
+    for chat in chats:
+      chatMap[chat.id] = chat
     for msg in messages:
-      for chat in chats:
-        if chat.id == msg.chatId:
-          self.events.emit(SIGNAL_SENDING_SUCCESS, MessageSendingSuccess(message: msg, chat: chat))
-          break
+      if chatMap.hasKey(msg.chatId):
+        let chat = chatMap[msg.chatId]
+        self.events.emit(SIGNAL_SENDING_SUCCESS, MessageSendingSuccess(message: msg, chat: chat))
+
+  proc processMessageUpdateAfterSend*(self: Service, response: RpcResponse[JsonNode]): (seq[ChatDto], seq[MessageDto]) =
```

Comment:
I know it's not your code, but this `AfterSend` is quite weird. 
The name of the function should describe what it does, not when it will be called. 😄 

```suggestion
  proc processMessageUpdateResponse*(self: Service, response: RpcResponse[JsonNode]): (seq[ChatDto], seq[MessageDto]) =
```

Reply:
Yes, that makes sense!
Since `protocol/messenger.go` uses the `MessengerResponse` type, it might be better to use the same name in proc. Something like `proc processMessengerResponse*(self: Service, response: RpcResponse[JsonNode]): `.